packages/datadog-plugin-kafkajs/src/consumer.js

```diff
@@ -20,10 +23,18 @@ class KafkajsConsumerPlugin extends ConsumerPlugin {
       metrics: {
         'kafka.partition': partition
       }
-    })
+    }
+
+    if (this.config.dsmEnabled) {
+      this.tracer.decodeDataStreamsContext(message.headers['dd-pathway-ctx'])
```

Comment:
Why is this exposed by the tracer? Is this a public API? Also, the current implementation is Kafka-specific, so is this really a generic API?

Reply:
I don't think that this is intended to be public. We will refactor it to make it no longer a method on the tracer object.